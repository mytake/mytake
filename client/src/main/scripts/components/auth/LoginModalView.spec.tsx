/*
 * MyTake.org website and tooling.
 * Copyright (C) 2020 MyTake.org, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * You can contact us at team@mytake.org
 */
import React from "react";
import renderer from "react-test-renderer";
import LoginModalView, { LoginModalViewEvents } from "./LoginModalView";

jest.mock("react-modal", () => ({
  __esModule: true,
  default: "Modal",
}));

const events: LoginModalViewEvents = {
  onFormSubmit: jest.fn(),
  onRequestClose: jest.fn(),
};

test("LoginModalView open", () => {
  const tree = renderer
    .create(<LoginModalView isOpen={true} events={events} showForm={true} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("LoginModalView closed", () => {
  const tree = renderer
    .create(<LoginModalView isOpen={false} events={events} showForm={true} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("LoginModalView open with response", () => {
  const tree = renderer
    .create(
      <LoginModalView
        isOpen={true}
        events={events}
        loginRes={{
          title: "Title Text",
          body: "Lorem ipsum body.",
          btn: "Button Text",
        }}
        showForm={false}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("LoginModalView open with response and form", () => {
  const tree = renderer
    .create(
      <LoginModalView
        isOpen={true}
        events={events}
        loginRes={{
          title: "Login Required",
          body: "Your login timed out.",
        }}
        showForm={true}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
