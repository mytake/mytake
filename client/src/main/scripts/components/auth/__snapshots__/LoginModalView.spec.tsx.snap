// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginModalView closed 1`] = `
<Modal
  className="modal"
  isOpen={false}
  onRequestClose={[MockFunction]}
  overlayClassName="modal__overlay"
  portalClassName="modal__portal"
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={false}
>
  <form
    className="modal__form"
    onSubmit={[Function]}
  >
    <label
      className="modal__label"
      htmlFor="email"
    >
      Email
    </label>
    <input
      className="modal__input"
      id="email"
      onChange={[Function]}
      required={true}
      type="email"
      value=""
    />
    <input
      className="modal__button"
      type="submit"
      value="Login or Create Account"
    />
  </form>
  <button
    className="modal__close"
    onClick={[MockFunction]}
  >
    <svg
      fill="none"
      height={24}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="18"
        x2="6"
        y1="6"
        y2="18"
      />
      <line
        x1="6"
        x2="18"
        y1="6"
        y2="18"
      />
    </svg>
  </button>
</Modal>
`;

exports[`LoginModalView open 1`] = `
<Modal
  className="modal"
  isOpen={true}
  onRequestClose={[MockFunction]}
  overlayClassName="modal__overlay"
  portalClassName="modal__portal"
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={false}
>
  <form
    className="modal__form"
    onSubmit={[Function]}
  >
    <label
      className="modal__label"
      htmlFor="email"
    >
      Email
    </label>
    <input
      className="modal__input"
      id="email"
      onChange={[Function]}
      required={true}
      type="email"
      value=""
    />
    <input
      className="modal__button"
      type="submit"
      value="Login or Create Account"
    />
  </form>
  <button
    className="modal__close"
    onClick={[MockFunction]}
  >
    <svg
      fill="none"
      height={24}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="18"
        x2="6"
        y1="6"
        y2="18"
      />
      <line
        x1="6"
        x2="18"
        y1="6"
        y2="18"
      />
    </svg>
  </button>
</Modal>
`;

exports[`LoginModalView open with response 1`] = `
<Modal
  className="modal"
  isOpen={true}
  onRequestClose={[MockFunction]}
  overlayClassName="modal__overlay"
  portalClassName="modal__portal"
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={false}
>
  <h2
    className="modal__header"
  >
    Title Text
  </h2>
  <p
    className="modal__text"
  >
    Lorem ipsum body.
  </p>
  <button
    className="modal__button"
    onClick={[MockFunction]}
  >
    Button Text
  </button>
  <button
    className="modal__close"
    onClick={[MockFunction]}
  >
    <svg
      fill="none"
      height={24}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="18"
        x2="6"
        y1="6"
        y2="18"
      />
      <line
        x1="6"
        x2="18"
        y1="6"
        y2="18"
      />
    </svg>
  </button>
</Modal>
`;

exports[`LoginModalView open with response and form 1`] = `
<Modal
  className="modal"
  isOpen={true}
  onRequestClose={[MockFunction]}
  overlayClassName="modal__overlay"
  portalClassName="modal__portal"
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={false}
>
  <h2
    className="modal__header"
  >
    Login Required
  </h2>
  <p
    className="modal__text"
  >
    Your login timed out.
  </p>
  <form
    className="modal__form"
    onSubmit={[Function]}
  >
    <label
      className="modal__label"
      htmlFor="email"
    >
      Email
    </label>
    <input
      className="modal__input"
      id="email"
      onChange={[Function]}
      required={true}
      type="email"
      value=""
    />
    <input
      className="modal__button"
      type="submit"
      value="Login or Create Account"
    />
  </form>
  <button
    className="modal__close"
    onClick={[MockFunction]}
  >
    <svg
      fill="none"
      height={24}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="18"
        x2="6"
        y1="6"
        y2="18"
      />
      <line
        x1="6"
        x2="18"
        y1="6"
        y2="18"
      />
    </svg>
  </button>
</Modal>
`;
